### PR TITLE
Add MacSDK test script 

### PR DESCRIPTION
**Pull request by alexischr**

### CODE REVIEW

File: scripts/ci/run-jenkins.sh [hunk 1]
```diff
@@ -105,7 +105,10 @@ if [[ ${CI_TAGS} == *'monolite'* ]]; then make get-monolite-latest; fi
 make_parallelism=-j4
 if [[ ${label} == 'debian-8-ppc64el' ]]; then make_parallelism=-j1; fi
 
-${TESTCMD} --label=make --timeout=${make_timeout} --fatal make ${make_parallelism} -w V=1
+if [[ ${CI_TAGS} != *'mac-sdk'* ]]; # Mac SDK builds Mono itself
+	then
```
Comment (akoeplinger) [inline scripts/ci/run-jenkins.sh]: it probably makes more sense to move this check right at the top of the file so we can skip autogen.sh etc. as well and directly call into run-test-mac-sdk.sh:

```bash 
if [[ ${CI_TAGS} = *'mac-sdk'* ]]; # Mac SDK builds Mono itself
 	then
	${MONO_REPO_ROOT}/scripts/ci/run-test-mac-sdk.sh
	exit $?
fi
```

File: scripts/ci/run-test-mac-sdk.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# switch to using package Mono instead of system
+export PATH=${MONO_REPO_ROOT}/external/bockbuild/stage/bin:$PATH
+
+# Bundled MSBuild
+cd ${MONO_REPO_ROOT}/external/bockbuild/builds/msbuild-15.4/
+${TESTCMD} --label="msbuild-tests" --timeout=180m ./cibuild.sh --scope Test --host Mono --target Mono
+
+# Bundled LLVM
+cd ${MONO_REPO_ROOT}/external/bockbuild/builds/mono
```
Comment (akoeplinger) [inline scripts/ci/run-test-mac-sdk.sh]: judging from the build it looks like this is the wrong dir

File: scripts/ci/run-jenkins.sh (hunk 1 above)
Comment (alexischr) [reply scripts/ci/run-jenkins.sh]: I did it after autogen.sh on purpose as it takes care of inflating the submodules but I guess Jenkins can do that

Comment (akoeplinger) [reply scripts/ci/run-jenkins.sh]: yes it does :)

Comment (akoeplinger) [reply scripts/ci/run-jenkins.sh]: and bockbuild runs autogen.sh either doesn't it?

**Pull request merged**